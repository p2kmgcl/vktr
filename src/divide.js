import isNotEqualTo from "./assertions/is-not-equal-to";
import isNumber from "./assertions/is-number";
import isVector from "./assertions/is-vector";
import map from "./map";

/**
 * Returns a new Vector result of (a / scalar)
 * @param {Vector} a
 * @param {number} scalar
 * @return {Vector}
 */
function divide(a, scalar) {
  isVector(a);
  isNumber(scalar);
  isNotEqualTo(scalar, 0);

  return map(a, component => component / scalar);
}

export default divide;
