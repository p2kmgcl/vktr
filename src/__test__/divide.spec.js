import divide from "../divide";
import Vector from "../vector";

describe("divide()", () => {
  it("divides every component by a number", () => {
    const a = new Vector(0, 1, 0, -2);

    expect(divide(a, -4)).toMatchSnapshot();
  });

  it("requires a vector parameter", () => {
    expect(() => divide(null)).toThrow(TypeError);
    expect(() => divide("a")).toThrow(TypeError);
    expect(() => divide([1])).toThrow(TypeError);
    expect(() => divide(NaN)).toThrow(TypeError);
  });

  it("requires a number parameter", () => {
    const a = new Vector(1);

    expect(() => divide(a, null)).toThrow(TypeError);
    expect(() => divide(a, "a")).toThrow(TypeError);
    expect(() => divide(a, [1])).toThrow(TypeError);
    expect(() => divide(a, NaN)).toThrow(TypeError);
  });

  it("requires a non-zero number", () => {
    const a = new Vector(1);

    expect(() => divide(a, 0)).toThrow(RangeError);
  });
});
