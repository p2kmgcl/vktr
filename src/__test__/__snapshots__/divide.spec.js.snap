// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`divide() divides every component by a number 1`] = `
Vector {
  "components": Array [
    -0,
    -0.25,
    -0,
    0.5,
  ],
}
`;
